File: tests/fixtures/edgecase_repo/kotlin/Service6.kt
package com.example.gen

import com.example.gen.Base6

interface Handler6 {
    fun handle(payload: String): Boolean
}

open class Base6(val name: String)

class Service6(name: String) : Base6(name), Handler6 {
    override fun handle(payload: String): Boolean {
        validate(payload)
        return true
    }

    private fun validate(payload: String) {}
}

fun helper6(value: String): String {
    return value
}

File: tests/fixtures/edgecase_repo/csharp/Service6.cs
using System;

namespace Example.Gen
{
    public interface IHandler6
    {
        bool Handle(string payload);
    }

    public class Base6
    {
        protected string Name;
    }

    public class Service6 : Base6, IHandler6
    {
        public Service6(string name)
        {
            this.Name = name;
        }

        public bool Handle(string payload)
        {
            this.Validate(payload);
            return true;
        }

        private void Validate(string payload) { }
    }
}
